src/System.Private.CoreLib/shared/System/Threading/Mutex.Windows.cs

```diff
@@ -21,18 +21,10 @@ public sealed partial class Mutex : WaitHandle
 #if PLATFORM_UNIX
         // Maximum file name length on tmpfs file system.
         private const int WaitHandleNameMax = 255;
+#else
+        private const int WaitHandleNameMax = Interop.Kernel32.MAX_PATH;
```

Comment:
This number is not going to be right on any recent Windows; and it is a best guess on Unix as well. I think we should remove the length from the message.